packages/features/ee/payments/components/Payment.tsx

```diff
@@ -53,21 +51,35 @@ const PaymentForm = (props: Props) => {
 
     let payload;
     const params: { [k: string]: any } = {
-      uid: props.bookingUid,
+      uid: props.booking.uid,
       email: router.query.email,
     };
+
+    const returnUrl = new URL(
```

Comment:
Can we encapsulate the return URL logic in a function?

```suggestion
    const return_url = getReturnUrl(props);
```